### PR TITLE
added some tests with gherkins

### PR DESCRIPTION
**Pull request by nicolasvallejo13**
Feature: Trello Members API # boards.feature:1
  As a user I want to use Boards API and perform Get,Post And Delete operations
  Scenario: GET boards of id                                      # boards.feature:4
    Given the user defines a "GET" request to "members/me/boards" # steps/boards_steps.py:11
    When the user sends the request                               # steps/boards_steps.py:23
    Then verify the status code is "200"                          # steps/boards_steps.py:33
    And verify the user recives a "7" lengt item                  # steps/boards_steps.py:42

  Scenario: Get all information of board                                         # boards.feature:11
    Given the user defines a "GET" request to "/boards/623cd8eef6a5ea0970e67c12" # steps/boards_steps.py:11
    When the user sends the request                                              # steps/boards_steps.py:23
    Then verify the status code is "200"                                         # steps/boards_steps.py:33
    And verify the user recives a "12" lengt item                                # steps/boards_steps.py:42

  Scenario: GET memebers of board                                                        # boards.feature:18
    Given the user defines a "GET" request to "/boards/623cd8eef6a5ea0970e67c12/members" # steps/boards_steps.py:11
    When the user sends the request                                                      # steps/boards_steps.py:23
    Then verify the status code is "200"                                                 # steps/boards_steps.py:33
    And verify there is at least 1 member                                                # steps/boards_steps.py:37

  Scenario: Updating information of a Board                                      # boards.feature:24
    Given the user defines a "PUT" request to "/boards/6287b012e56fca6695059571" # steps/boards_steps.py:11
      | Key  | Value              |
      | name | testingnicolasv    |
      | desc | a valid data table |
    When the user sends the request                                              # steps/boards_steps.py:23
    Then verify the status code is "200"                                         # steps/boards_steps.py:33
    And verify the new object contains the following info                        # steps/boards_steps.py:49
      | Key  | Value              |
      | name | testingnicolasv    |
      | desc | a valid data table |

  Scenario: Updating information of a Board and verify the board                 # boards.feature:36
    Given the user defines a "PUT" request to "/boards/6287b012e56fca6695059571" # steps/boards_steps.py:11
      | Key  | Value              |
      | name | testingnicolasv    |
      | desc | a valid data table |
    When the user sends the request                                              # steps/boards_steps.py:23
    Then verify the status code is "200"                                         # steps/boards_steps.py:33
    And verify the board contains the following info                             # steps/boards_steps.py:93
      | Key  | Value              |
      | name | testingnicolasv    |
      | desc | a valid data table |

  Scenario: CREATE and DELETE a board                     # boards.feature:48
    Given the user defines a "POST" request to "/boards"  # steps/boards_steps.py:11
      | Key  | Value              |
      | name | testingnicolasv    |
      | desc | a valid data table |
    When the user sends the request                       # steps/boards_steps.py:23
    Then verify the status code is "200"                  # steps/boards_steps.py:33
    And verify the new object contains the following info # steps/boards_steps.py:49
      | Key  | Value              |
      | name | testingnicolasv    |
      | desc | a valid data table |
    And DELETE a Board                                    # steps/boards_steps.py:59

  Scenario: CREATE and DELETE a board2                    # boards.feature:63
    Given the user defines a create_board function        # steps/boards_steps.py:64
      | Key  | Value       |
      | name | kiarajose12 |
    When the user calls the create_board function         # steps/boards_steps.py:73
    Then verify the status code is "200"                  # steps/boards_steps.py:33
    And verify the new object contains the following info # steps/boards_steps.py:49
      | Key  | Value       |
      | name | kiarajose12 |
    And DELETE a Board                                    # steps/boards_steps.py:59

  Scenario: Duplicate a Board and DELETE it                                                               # boards.feature:77
    Given the user defines a duplicate function with id "6287b86ed19bf5054179cb2f" and new name "NewName" # steps/boards_steps.py:82
    When the user sends the copy_board function                                                           # steps/boards_steps.py:87
    Then verify the status code is "200"                                                                  # steps/boards_steps.py:33
    And DELETE a Board                                                                                    # steps/boards_steps.py:59

  Scenario: GET actions of board                                                         # boards.feature:84
    Given the user defines a "GET" request to "/boards/623cd8eef6a5ea0970e67c12/actions" # steps/boards_steps.py:11
    When the user sends the request                                                      # steps/boards_steps.py:23
    Then verify the status code is "200"                                                 # steps/boards_steps.py:33
    And verify the user recives a "50" lengt item                                        # steps/boards_steps.py:42

  Scenario: GET checklist of board                                                          # boards.feature:90
    Given the user defines a "GET" request to "/boards/627c0cb01e98336259fa1ae2/checklists" # steps/boards_steps.py:11
    When the user sends the request                                                         # steps/boards_steps.py:23
    Then verify the status code is "200"                                                    # steps/boards_steps.py:33
    And verify the user recives a "1" lengt item                                            # steps/boards_steps.py:42

  Scenario: GET labels of board                                                         # boards.feature:98
    Given the user defines a "GET" request to "/boards/627c0cb01e98336259fa1ae2/labels" # steps/boards_steps.py:11
    When the user sends the request                                                     # steps/boards_steps.py:23
    Then verify the status code is "200"                                                # steps/boards_steps.py:33
    And verify the user recives a "9" lengt item                                        # steps/boards_steps.py:42

  Scenario: CREATE label of board                                                        # boards.feature:105
    Given the user defines a "POST" request to "/boards/627c0cb01e98336259fa1ae2/labels" # steps/boards_steps.py:11
      | Key   | Value       |
      | name  | kiarajose12 |
      | color | orange      |
    When the user sends the request                                                      # steps/boards_steps.py:23
    Then verify the status code is "200"                                                 # steps/boards_steps.py:33
    And verify the new object contains the following info                                # steps/boards_steps.py:49
      | Key   | Value       |
      | name  | kiarajose12 |
      | color | orange      |

1 feature passed, 0 failed, 0 skipped
12 scenarios passed, 0 failed, 0 skipped
50 steps passed, 0 failed, 0 skipped, 0 undefined
Took 0m4.074s

### CODE REVIEW

File: tests/api/features/boards.feature [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+            | name        | kiarajose12            |
+        And DELETE a Board
+
+    
+    
+    Scenario: Duplicate a Board and DELETE it
+        Given the user defines a duplicate function with id "6287b86ed19bf5054179cb2f" and new name "NewName"
+        When the user sends the copy_board function
+        Then verify the status code is "200"
+        And DELETE a Board
```
Comment (atbootcamp) [inline tests/api/features/boards.feature:81]: - avoid conjunctive steps (AND)
- copy_board??? 
the flow and even the code tends to be a mess when using technical vs. functional or behavioral style statements

example of technical statements
```suggestion
        Given the user defines "POST" request to "/members/me/boards"
         | Parameter        |  Value  |
         | idBoardSource  | 6287b86ed19bf5054179cb2f |
         | name                | NewName |
        When the user sends the request
        Then verify the status code is "200"
        When the user defines "GET" request to "/boards/6287b86ed19bf5054179cb2f"
        And the user sends the request
        Then verify the status code is "200"
        And verify the source board and new board data matches except by
                 | params|
                 | id  |
                 | name | 
        When the user defines "DELETE" request to "/boards/{boardID}"
        And the user sends the request
        Then verify the status code is "200"
        When the user defines "GET" request to "/boards/{boardID}"
        And the user sends the request
        Then verify the status code is "404"
```

behavioral style
``` suggestion
       Given the user created a board with following data
                | Parameter        |  Value  |
                | idBoardSource  | 6287b86ed19bf5054179cb2f |
                | name                 | NewName                  |
        When the user retrieved the source board "6287b86ed19bf5054179cb2f"
        Then verify the source board and new board data matches except by
                 | params|
                 | id  |
                 | name |                  
        When the user deletes the board "NewName"
        Then verify the board "NewName" cannot be found
```

File: tests/api/features/boards.feature [hunk 2]
```diff
@@ -0,0 +1,119 @@
+Feature: Trello Members API
+    As a user I want to use Boards API and perform Get,Post And Delete operations
+
+    Scenario: GET boards of id
```
[the remark this answers is not in the document]
Comment (nicolasvallejo13) [reply tests/api/features/boards.feature:4]: how can I improve it?

File: tests/api/features/boards.feature (hunk 1 above)
Comment (nicolasvallejo13) [reply tests/api/features/boards.feature:81]: What do you men with "avoid conjunctive steps (AND)"  I mean, when should we use And when not? I don't catch the logic, shouldn't that be used after a When or give step?

File: tests/api/features/boards.feature (hunk 2 above)
[the remark this answers is not in the document]
Comment (atbootcamp) [reply tests/api/features/boards.feature:4]: the title of the scenario has to reflect the purpose/goal of the tests
so according to the steps it seems that it should be
```
    Scenario: GET members on board
    or
    Scenario: GET "n" members on board
```

**Pull request closed without merging**